backend/wrapped/serializers.py

```diff
@@ -0,0 +1,105 @@
+from rest_framework import serializers
+from .models import Page, IndividualStat, GlobalStat, IndividualStatThrough, GlobalStatThrough, Semester, User
+
+class IndividualStatSerializer(serializers.ModelSerializer):
+    key = serializers.StringRelatedField()
+
+    class Meta:
+        model = IndividualStat
+        fields = ['key', 'value', 'semester']
+
+
+class GlobalStatSerializer(serializers.ModelSerializer):
+    key = serializers.StringRelatedField()
+
+    class Meta:
+        model = GlobalStat
+        fields = ['key', 'value', 'semester']
+
+
+class IndividualStatThroughSerializer(serializers.ModelSerializer):
+    individual_stat_value = serializers.SerializerMethodField()
+
+    class Meta:
+        model = IndividualStatThrough
+        fields = ['text_field_name', 'individual_stat_value']
+
+    def get_individual_stat_value(self, obj):
+        user = self.context.get('user')
+        semester = self.context.get('semester')
+
+        try:
+            individual_stat = IndividualStat.objects.get(
+                user=user,
+                key=obj.IndividualStatKey,
+                semester=semester
+            )
+            return individual_stat.value
+        except IndividualStat.DoesNotExist:
+            return None
+
+
+class GlobalStatThroughSerializer(serializers.ModelSerializer):
+    global_stat_value = serializers.SerializerMethodField()
+
+    class Meta:
+        model = GlobalStatThrough
+        fields = ['text_field_name', 'global_stat_value']
+
+    def get_global_stat_value(self, obj):
+        semester = self.context.get('semester')
+
+        try:
+            global_stat = GlobalStat.objects.get(
+                key=obj.GlobalStatKey.key,
+                semester=semester
+            )
+            return global_stat.value
+        except GlobalStat.DoesNotExist:
+            return None
+
+
+class PageSerializer(serializers.ModelSerializer):
+    combined_stats = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Page
+        fields = ['name', 'template_path', 'combined_stats']
+
+    def get_combined_stats(self, obj):
+        user = self.context.get('user')
+        semester = self.context.get('semester')
+
+        individual_throughs = IndividualStatThrough.objects.filter(Page=obj)
```

Comment:
You a bit too much manual serialization here. As an outline the following is a bit more reasonable:
1. Serialize IndividualEntries to become a list of (field_name, value) pairs (using a serializer)
2.Serialize IndividualEntries to become a list of (field_name, value) pairs (using a serializer)
3. Combine into a dict and return. 

Although this seems hard to describe over a code review. Maybe we can sit down and peer program for this part